### PR TITLE
Adds parachain staking grant proposal

### PR DESCRIPTION
**Pull request by crystalin**
# Grant Application Checklist

- [x] The [application template](https://github.com/w3f/Open-Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [x] A BTC or Ethereum (DAI) address for the payment of the milestones is provided inside the application.  
- [x] I have read and acknowledged the [Terms and Conditions](https://github.com/w3f/Open-Grants-Program/blob/master/src/T&Cs.md).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The total funding amount of the project is below USD $30k for initial grant applications and $100k for follow-up grants.
- [x] The initial PR contains only one commit (squash if needed before submitting your PR).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery).

### CODE REVIEW

**Review by mmagician — CHANGES_REQUESTED** [review]
@crystalin Thanks for the application, looks pretty good. The only change that I'd like to request is changing the currency, because as of recently we only pay in DAI for Open Grants.

File: applications/parachain-staking.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+Antoine has been developing Web3 Dapps since 2016. He has participated in 3 projects inside ConsenSys (VariabL, Localties and Fathom) and helped HyperNetwork build their token system. Now, he's working for Moonbeam and has contributed to the polkadot-js suite (apps, common, api, etc).
+
+Joshy and Amar worked at Parity on the Substrate Recipes.
+
+## Development Roadmap :nut_and_bolt:
+
+* **Total Estimated Duration:** 6 weeks
+* **Full-time equivalent (FTE):** 2 FTE
+* **Total Costs:** 0.6 BTC
```
Comment (4meta5) [inline applications/parachain-staking.md:44]:
```suggestion
* **Total Costs:** 30k DAI
```

File: applications/parachain-staking.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+Both milestones will be worked on in parallel, but UI development will lag changes to the pallet implementation.
+
+### Milestone 1: Improve Parachain-Staking Pallet
+
+* **Estimated Duration:** 4 weeks 
+* **Costs:** 0.4 BTC
```
Comment (4meta5) [inline applications/parachain-staking.md:51]:
```suggestion
* **Costs:** 20k DAI
```

File: applications/parachain-staking.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+| 1. | Benchmark | benchmark the existing implementation to derive transaction weights and discern optimization opportunities |  
+| 2.  | Refactor to Minimal delegated Proof of Stake | Add constraint of max 1 nomination per `AccountId` and integrate `frame/pallet-proxy` to maintain support for multiple nominations. |
+| 3.  | Configurable Inflation | Replace sudo with governance origin for setting inflation rate. Provide instructions for replacing constant inflation with `pallet-staking`'s reward curve logic. | 
+| 4.  | Implement Slashing | Add equivocation detection to [`frame/aura`](https://github.com/paritytech/substrate/tree/master/frame/aura). Report slashing to `parachain-staking` and enforce prior to reward distribution. | 
+| 5.  | Docs | Update Rust crate-level docs and [user-level docs](https://meta5.world/stake-docs/) with implementation details. | 
+
+### Milestone 2: Parachain-Staking Polkadot-JS UI
+
+* **Estimated Duration:** 4 weeks 
+* **Costs:** 0.2 BTC
```
Comment (4meta5) [inline applications/parachain-staking.md:74]:
```suggestion
* **Costs:** 10k DAI
```

File: applications/parachain-staking.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
+# Open Grant Proposal
+
+> This document is referenced in the terms and conditions and therefore needs to contain all the required information. Don't remove any of the mandatory parts presented in bold letters or as headlines! See the [Open Grants Program Process](https://github.com/w3f/Open-Grants-Program/blob/master/README_2.md) on how to submit a proposal.
+
+* **Project:** Pallet-dPoS for Parachain Staking
+* **Proposer:** [Moonbeam Network](https://github.com/PureStake/moonbeam)
+* **Payment Address:** 1EMk3z9VMmXN6y3qwA37s6Qps4xRKGGE9G
```
Comment (4meta5) [inline applications/parachain-staking.md:7]: @crystalin this needs to changed to a DAI address as well

**Comment by crystalin** [comment]
The PR is not updating after pushing changes to the branch, I'm closing/re-opening the PR

**Comment by crystalin** [comment]
@mmagician I've updated the currency/address

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. Could you potentially also add the deliveries 0a to 0d to the application, see https://github.com/w3f/Open-Grants-Program/blob/master/applications/application-template.md#milestone-1-example--implement-substrate-modules. At least the license should be included. Regarding milestone 2, would it potentially be possible to also make a pr to integrate this into polkadot.js apps itself?

**Comment by 4meta5** [comment]
@Noc2 I made the requested changes, let me know if they are OK. I can change the licenses if necessary.

**Review by mmagician — APPROVED** [review]
Looks good to me, I'm looking forward to see this integrated!

**Review by Noc2 — APPROVED** [review]

**Review by alxs — APPROVED** [review]

**Comment by alxs** [comment]
Hey @crystalin @4meta5, care to share a quick status update on your second milestone?

**Comment by alxs** [comment]
@crystalin sounds good, thanks for the update. Please keep us posted.

**Comment by 4meta5** [comment]
@alxs 2 updates:
1. There was a recently significant refactor of the **1st milestone** `parachain-staking` pallet in https://github.com/PureStake/moonbeam/pull/810 . This PR changed the naming conventions from NPoS to DPoS. It also optimized the exits so that they are no longer scheduled to be automatically executed in `on_initialize` (which had led to large (sometimes overweight) blocks at round transitions). These changes are included in the upcoming NCC audit.
2. The **2nd milestone** work is complete https://github.com/polkadot-js/apps/pull/6430 but the PR is awaiting review. We are waiting for it to be approved and merged before formally submitting the milestone.

**Pull request merged**